app/controllers/api/v1/deliveries_controller.rb

```diff
@@ -124,6 +125,10 @@ def serializer
       Api::V1::DeliverySerializer
     end
 
+    def delete_existing_delivery
+      Delivery.where(offer_id: params[:delivery][:offer_id]).map(&:destroy)
```

Comment:
@swatijadhav If ```params[:delivery][:offer_id]``` evaluates to ```nil``` then we might have unexpected consequences here. Can you add a check to ensure it's not nil? Thanks.